miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -505,6 +510,16 @@ private void loadChildEntities(Sample sample) throws IOException {
       if (sai.getSubproject() != null && sai.getSubproject().getId() != null) {
         sai.setSubproject(subProjectDao.getSubproject(sai.getSubproject().getId()));
       }
+      if (isTissueProcessingSample(sai) && sai instanceof SampleSlide) {
+        Stain originalStain = ((SampleSlide) sai).getStain();
+        Stain stain;
+        if (originalStain == null) {
+          stain = null;
+        } else {
+          stain = stainDao.get(originalStain.getId());
```

Comment:
I don't know. We seem to have no set pattern as to whether a service should access a service or a DAO.